### PR TITLE
[Merged by Bors] - Update to kube-rs 0.82.2 and dependencies

### PR DESCRIPTION
**Pull request by maltesander**
## Description

fixes https://github.com/stackabletech/operator-rs/issues/588

updated dependencies:
- kube-rs 0.82.2
- k8s-openapi 0.18.0
- json_patch 1.0.0
- rstest 0.17.0

other:
- derive defaults for some enums instead of impl (clippy)

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -426,34 +426,35 @@ impl Client {
...
     ///          Duration::from_secs(1),
-    ///          client.wait_created::<Pod>(&client.default_namespace, lp.clone()),
+    ///          client.wait_created::<Pod>(&client.default_namespace, watcher_config),
     ///      )
     ///      .await;
     /// }
     /// ```
     ///
-    pub async fn wait_created<T>(&self, namespace: &T::Namespace, lp: ListParams)
+    pub async fn wait_created<T>(&self, namespace: &T::Namespace, watcher_config: watcher::Config)
```
Comment (razvan) [inline src/client.rs:451]: How about not changing the signature so as to not introduce an unnecessary incompatibility?

Comment (maltesander) [reply src/client.rs:451]: I dont understand?

The watcher now gets a different struct (granted, some fields are similar) but there are differences like https://github.com/kube-rs/kube/blob/e01187e13ba364ccecec452e023316a62fb13e04/kube-runtime/src/watcher.rs#L206 that should be configurable?

You just want to use the Label / Field selectors from the ListParams? We will be missing config options by doing this.

**Review by lfrancke — APPROVED** [review]
It looks good to me and I'll approve but maybe someone else, closer to the code, wants to take a look as well.

**Review by razvan — APPROVED** [review]
lgtm

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**